### ISSUE TITLE
Installing the pry gem on Windows fails starting with rubygems 2.4.5

### ISSUE THREAD

**Issue by ryanatball** [rubygems/rubygems#1120]
```
gem update --system 2.4.5
gem install pry
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    bad response Not Found 404 (https://bb-m.rubygems.org/quick/Marshal.4.8/pry-0.10.1-x86-mingw32.gemspec.rz)

gem update ---system 2.4.4
gem install pry
Successfully installed pry-0.10.1-x86-mingw32
Parsing documentation for pry-0.10.1-x86-mingw32
1 gem installed
```

**Comment by ryanatball** [comment]
I suspect this has something to do with the platform.   The above URL does not work:
https://bb-m.rubygems.org/quick/Marshal.4.8/pry-0.10.1-x86-mingw32.gemspec.rz

But this one does:
https://bb-m.rubygems.org/quick/Marshal.4.8/pry-0.10.1-i386-mingw32.gemspec.rz

Perhaps earlier rubygems were more lenient with platform?

**Comment by tenderlove** [comment]
I don't have a windows machine to test on.  Is there any way you can bisect this problem?  You should be able to check out the RG git repo and just do `ruby -I lib bin/gem install pry` to test on each revision.

**Comment by ryanatball** [comment]
I'll see if I can narrow down to the commit tomorrow.   Thanks for the response.

**Comment by ryanatball** [comment]
```
c:\git\rubygems>git checkout 252c06c
Previous HEAD position was d96c57c... Convert dependency API platforms upon load
HEAD is now at 252c06c... Hide internal method prefetch_now

c:\git\rubygems>ruby -I lib bin/gem install pry
Successfully installed pry-0.10.1-x86-mingw32
Parsing documentation for pry-0.10.1-x86-mingw32
Done installing documentation for pry after 1 seconds
1 gem installed

c:\git\rubygems>git checkout d96c57c
Previous HEAD position was 252c06c... Hide internal method prefetch_now
HEAD is now at d96c57c... Convert dependency API platforms upon load

c:\git\rubygems>ruby -I lib bin/gem install pry
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    bad response Not Found 404 (https://bb-m.rubygems.org/quick/Marshal.4.8/pry-0.10.1-x86-mingw32.gemspec.rz)
```

**Comment by ryanatball** [comment]
Looks like it broke starting at d96c57c

**Comment by oddityoverseer13** [comment]
I'm also seeing this issue:

```
PS C:\> gem install pry
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    bad response Not Found 404 (http://rubygems.global.ssl.fastly.net/quick/Marshal.4.8/pry-
0.10.1-x86-mingw32.gemspec.rz)
```

**Comment by jnyman** [comment]
Just to confirm, I too see this issue:

    $ gem install pry
    ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
        bad response Not Found 404 (https://rubygems.global.ssl.fastly.net/quick/Marshal.4.8/pry-0.10.1-x86-mingw32.gemspec.rz)

Is there a way to substitute in the URL that ryanatball mentioned above? I wouldn't mind trying to take a past version that worked, but pry is being installed as a dependency with Chef so I don't know if there will be any incompatibilities.

**Comment by drbrain** [comment]
This was broken by rubygems/rubygems#1058.

**Comment by drbrain** [comment]
#1058 normalizes the platform (via Gem::Platform.new) which is wrong.

The SpecFetcher API matched the local, parsed platform against a String platform which worked OK.  The ApiSpecification needs to be able to match the its platform against the required platform later on.

**Comment by jdmundrawala** [comment]
Thoughts on https://github.com/rubygems/rubygems/pull/1184?

**Comment by dpneumo** [comment]
Any plans to fix this? Am on Win 7. Using gem update --system  2.4.4 allows gem install pry to work. However, seems less than ideal.

**Comment by BBlackwo** [comment]
Thanks ryanatball! Installing Gem 2.4.4 fixed my issue :) I had 2.4.6.

**Comment by danShumway** [comment]
Has any progress been made in this area / is anyone working on it?

**Comment by Christianjuth** [comment]
I am getting this issue too

```
jenkins@Ruby-Slave:~/workspace/operation-bye-bye-empty-companies$ gem install nutshell-crm-api
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    bad response Not Found 404 (https://rubygems.global.ssl.fastly.net/gems/nutshell-crm-api-0.1.01.gem)
```

I would also like to note that the lastest version of the gem I am trying to install is 0.1.1 and not 0.1.01 so I don't know what the heck is up with that.

**Comment by konung** [comment]
Same issue using rubygems 2.4.8. Downgrading to 2.4.4 worked. Using windows 7 64 bit with Ruby 2.2.3 32 bit

**Comment by cablegunmaster** [comment]
Still occuring issue, downgraded to 2.4.4 aswell and now it works.

**Comment by strcmp** [comment]
pry v0.10.3 was released and it should fix the issue. we weren't able to test on windows because we didn't have a VM for it or a machine with it. can someone confirm it works with rubygems 2.4.8? cheers.

**Comment by tw0517tw** [comment]
@strcmp using gem 2.4.5.1 `gem install pry` installed pry 0.10.3 successfully.
but travis CLI use pry 0.9, so it will keep happening when people install https://github.com/travis-ci/travis.rb

**Comment by strcmp** [comment]
ah i see, hm good to know that it is fixed from 0.10.3 onwards but i guess you will have to wait for rubygems to fix this issue or ask travis-ci to upgrade the gem.

**Comment by djberg96** [comment]
I don't really see a good way to fix this because of the way this gem was created, i.e. explicitly setting platform to i386 and the arch getting set to x86. The latest release of pry fixes the issue, and it won't happen again with Rubygems 2.5, but I think we have to punt on this specific case.
